WordPress/src/main/java/org/wordpress/android/ui/comments/CommentAdapter.java

```diff
@@ -409,36 +470,36 @@ protected void onCancelled() {
         }
         @Override
         protected Boolean doInBackground(Void... params) {
-            if (mStatusFilter == null){
-                tmpComments = CommentTable.getCommentsForBlogWithFilter(mLocalBlogId, CommentStatus.UNKNOWN);
+            List<CommentModel> comments;
+            if (mStatusFilter == null || mStatusFilter == CommentStatus.ALL) {
+                // The "all" filter actually means "approved" + "unapproved" (but not "spam", "trash" or "deleted")
+                comments = mCommentStore.getCommentsForSite(mSite, CommentStatus.APPROVED, CommentStatus.UNAPPROVED);
             } else {
-                tmpComments = CommentTable.getCommentsForBlogWithFilter(mLocalBlogId, mStatusFilter);
+                comments = mCommentStore.getCommentsForSite(mSite, mStatusFilter);
             }
 
-            if (mComments.isSameList(tmpComments)) {
-                return false;
-            }
+            tmpComments = new CommentList();
+            tmpComments.addAll(comments);
 
-            // pre-calc transient values so they're cached prior to display
-            for (Comment comment: tmpComments) {
-                comment.getDatePublished();
-                comment.getUnescapedPostTitle();
-                comment.getAvatarForDisplay(mAvatarSz);
-                comment.getFormattedTitle();
-
-                String content = StringUtils.notNullStr(comment.getCommentText());
-                //to load images embedded within comments, pass an ImageGetter to WPHtml.fromHtml()
-                Spanned spanned = WPHtml.fromHtml(content, null, null, mContext, null, 0);
-                comment.setUnescapedCommentWithDrawables(spanned);
-            }
-
-            return true;
+            return !mComments.isSameList(tmpComments);
         }
         @Override
         protected void onPostExecute(Boolean result) {
             if (result) {
                 mComments.clear();
                 mComments.addAll(tmpComments);
+                // Sort by date
```

Comment:
> Not sure if it's more efficient, some sorting algorithms hit their absolute worst case

In this case, `Collections.sort()` should be using Timsort, which is most efficient for already sorted lists (I guess that might change in future Java versions, but it seems unlikely).

Adding an option in the getter seems the most flexible solution 👍.